### DISCUSSION TITLE
publish to netlify with params

### DISCUSSION THREAD

**Discussion by dsolito** [quarto-dev/quarto-cli#5310] [Q&A] [answered]
Hello,

Is it possible to deploy the qmd file to netlify with params?
I tried to use -P but doesn't work with the publish command.

quarto publish netlify document.qmd -P doc_code:"1234" 

Thank you.

**Comment by mcanouil** [comment, marked as answer]
No you can't.
<img width="1095" alt="image" src="https://user-images.githubusercontent.com/8896044/234540258-54fb16a2-e76a-4661-bf82-16315f9ff53c.png">

You'll have to use `quarto render` and handle the publishing part yourself for example as a post-render script, see <https://quarto.org/docs/projects/scripts.html#pre-and-post-render>.

This being said, maybe `quarto publish` could inherits from `quarto render` parameters.

**Reply by dsolito** [reply]
Thank you!
I was not aware of the post-render script.
This could be the solution for the {targets} pipeline.

David

**Reply by mcanouil** [reply]
Another option could be (not tested) to `quarto render document.qmd -P doc_code:"1234" && quarto publish --no-render netlify document.qmd`.
